src/coreclr/vm/jitinterface.cpp

```diff
@@ -13394,8 +13409,16 @@ PCODE UnsafeJitFunction(PrepareCodeConfig* config,
         printf(".");
 #endif // _DEBUG
 
-    FastInterlockExchangeAddLong((LONG64*)&g_cbILJitted, methodInfo.ILCodeSize);
-    FastInterlockIncrement((LONG*)&g_cMethodsJitted);
+    timer.Stop();
+
+    InterlockedExchangeAdd64((LONG64*)&g_c100nsTicksInJit, timer.Elapsed100nsTicks());
```

Comment:
You are doing the floating match to compute the ticks twice here. That is unnecessary.

Reply:
The timer caches the calculation and keeps track of whether it has been reset or lapped since the last time. Subsequent calls shouldn't recalculate.

Reply:
> The timer caches the calculation

Ok, I have not noticed that. This caching feels like an overengineering to me. It is trivial for the typical caller to call it just once.

Notice that the .NET Stopwatch type does not have a cache like that either.

Reply:
> This caching feels like an overengineering to me.

The intent is to be simple and reusable, so I'll pull that out.